### PR TITLE
Wire in pagination to LabeledSelect, use when installing Rancher Backup chart

### PR DESCRIPTION
**Pull request by richard-cox**
Covnerted to draft. This will be enabled via the core vai cache feature flag

### Summary
Contributes to https://github.com/rancher/dashboard/issues/9964

### Occurred changes and/or fixed issues
Update the LabelSelect component to handle large number of resources by making use of new pagination api
- New mixin (this was already written pre composite change) handles pagination functionality
- Mixin provides the hooks to LabelSelect to control and display pagination features (except fn to fetch pages)
- Components provide a `paginate` function to the LabelSelect (mixin) which when ran returns the page to show. This only handles the http request and nothing else. This is inplace of the owning component passing in `options`. 
  - In the future some of this can be made reusable by other components that will have paginated LabelSelects 

Apply the new paginate feature to the 
Add initial support for paginated requests in the request action pipeline
- This was copied over from another (draft) PR which will support list pagination

Other changes resulting from design review
- The 'group' visuals have been improved and can now show an icon
- Note - Colours have not been changed as per design. These all work with generic colours that apply to the whole dashboard. Changing them specifically for this component would introduce tech debt. We should consider the component colour when making changes to all colours.

### Technical notes summary
- LabelSelect has been moved to it's own folder to contain all LabelSelect associated files. This resulting in a change to some TS based imports
- Caching support for pagination requests has not yet been integrated, so every pagination request will result in requests from rancher to kube to fetch all resources (this can be slow). However using the new api will still be an improvement given the ui would fetch 20k+ resources (only once) but the initial load would be very slow... and the drop down would be unusable trying to show all 20k+. At the point caching is enabled this feature would then become super quick

### Areas or cases that should be tested
- General use - local cluster --> Cluster Tools --> Rancher Backup Install --> Next --> Use an S3-compatible object store --> Credential Secret
- Selecting multiple options - Logging Product --> Flows --> Create --> MAtches tab --> `Limit to specific nodes`
- Complex use case - Deployments --> create --> namespace drop down

At the least select and search should work for all above. For deployment creation ns drop down create should also work

### Areas which could experience regressions
- Anywhere we show drop downs

### Screenshot/Video
image

image

### CODE REVIEW

**Comment by richard-cox** [comment]
Linking design issue - UXUI-189

File: pkg/aks/components/AksNodePool.vue [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { mapGetters } from 'vuex';
 import { _CREATE, _VIEW } from '@shell/config/query-params';
 import type { AKSDiskType, AKSNodePool, AKSPoolMode } from '../types/index';
 import LabeledInput from '@components/Form/LabeledInput/LabeledInput.vue';
-import LabeledSelect from '@shell/components/form/LabeledSelect.vue';
+import LabeledSelect from '@shell/components/form/LabeledSelect/index.vue';
```
Comment (aalves08) [inline pkg/aks/components/AksNodePool.vue:8]: does this need the `index.vue`? on other places it wasn't needed

File: pkg/aks/components/CruAks.vue [hunk 2]
```diff
@@ -16,7 +16,7 @@ import CreateEditView from '@shell/mixins/create-edit-view';
 import FormValidation from '@shell/mixins/form-validation';
 import SelectCredential from '@shell/edit/provisioning.cattle.io.cluster/SelectCredential.vue';
 import CruResource from '@shell/components/CruResource.vue';
-import LabeledSelect from '@shell/components/form/LabeledSelect.vue';
+import LabeledSelect from '@shell/components/form/LabeledSelect/index.vue';
```
Comment (aalves08) [inline pkg/aks/components/CruAks.vue:19]: does this need the index.vue? on other places it wasn't needed

File: shell/components/form/__tests__/LabeledSelect.test.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { mount } from '@vue/test-utils';
-import LabeledSelect from '@shell/components/form/LabeledSelect.vue';
+import LabeledSelect from '@shell/components/form/LabeledSelect/index.vue';
```
Comment (aalves08) [inline shell/components/form/__tests__/LabeledSelect.test.ts:2]: same here

File: shell/components/form/LabeledSelect/labeled-select-pagination.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+  page: T[],
+  pages: number,
+  total: number
+}>
+
+interface Props {
+  paginate: PaginateFn
+}
+
+export default Vue.extend<Data, Methods, any, Props>({
```
Comment (aalves08) [inline shell/components/form/LabeledSelect/labeled-select-pagination.ts:50]: Shouldn't we be moving away of `Vue.extend` as per https://github.com/rancher/dashboard/issues/10104?

File: shell/components/form/LabeledSelect/labeled-select-pagination.ts [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+      type:    Function as PropType<PaginateFn>,
+    }
+  },
+
+  data(): Data {
+    return {
+      currentPage: 1,
+      search:      '',
+      pageSize:    10,
+
```
Comment (aalves08) [inline shell/components/form/LabeledSelect/labeled-select-pagination.ts:63]: nitpick, but blank line needed?

File: shell/chart/rancher-backup/index.vue [hunk 6]
```diff
@@ -113,6 +117,7 @@ export default {
         this.value.s3.enabled = true;
         break;
       default:
+
```
Comment (aalves08) [inline shell/chart/rancher-backup/index.vue:120]: another nitpick, but does it need blank line? code is usually right after the `switch` condition

File: pkg/aks/components/AksNodePool.vue (hunk 1 above)
Comment (richard-cox) [reply pkg/aks/components/AksNodePool.vue:8]: TS components need the explicit `index.vue`, JS is fine with just the folder name

File: pkg/aks/components/CruAks.vue (hunk 2 above)
Comment (richard-cox) [reply pkg/aks/components/CruAks.vue:19]: Same as https://github.com/rancher/dashboard/pull/10068#discussion_r1474680928

File: shell/components/form/__tests__/LabeledSelect.test.ts (hunk 3 above)
Comment (richard-cox) [reply shell/components/form/__tests__/LabeledSelect.test.ts:2]: Same as https://github.com/rancher/dashboard/pull/10068#discussion_r1474680928 (but this is a TS file)

File: shell/chart/rancher-backup/index.vue (hunk 6 above)
Comment (richard-cox) [reply shell/chart/rancher-backup/index.vue:120]: updated

**Review by aalves08 — APPROVED** [review]
Gave it another round of testing and LGTM!

Screenshot 2024-02-02 at 09 36 57

Ignore that `Vue.extend` comment for now since we are still in a pickle with the `defineComponent` malarky.

One other thing: That `istio` E2E has been flimsy these last few days... I've had to re-run that test in a lot of my PRs lately. Fortunately, one re-run was enough. :P

**Comment by richard-cox** [comment]
Superseded by https://github.com/rancher/dashboard/pull/10786

**Pull request closed without merging**